proto/viam/app/data/v1/data.proto

```diff
@@ -80,6 +80,14 @@ message Filter {
   repeated string org_ids = 11;
   repeated string mime_type = 12;
   CaptureInterval interval = 13;
+  TagsOption tags_option = 14;
+}
+
+enum TagsOption {
+  TAGS_OPTION_UNSPECIFIED = 0;
+  TAGS_OPTION_MATCHING = 1;
+  TAGS_OPTION_ALL = 2;
```

Comment:
Any thoughts on TAGGED/UNTAGGED/MATCHING_BY_OR or ALL_TAGGED/ALL_UNTAGGED/MATCHING_BY_OR?

See [here](https://github.com/viamrobotics/app/pull/901#discussion_r1034920632)

cc @AaronCasas for other opinions.

Reply:
But I agree that the naming is unclear, as it is

Reply:
Will there be a MATCH_BY_AND in the future?

Reply:
We are likely to add other operations on the array (AND, XOR) in the future, so want to futureproof here. 

+1 to TAGGED/UNTAGGED/MATCHING_BY_OR.

Reply:
Updated to be: TAGGED/UNTAGGED/MATCH_BY_OR

Reply:
Would we ever want to support a combo? e.g. (A & B) || C

Probably not something we have to deal with until there's a specific need, but if we would, this would probably need to be made more generalizable vs having separate rpcs for each logical operator

Reply:
Likely wouldn't need to worry about something that complex, but if we did, we would probably migrate to having a string passed in (e.g. `(tag.dog && tag.cat || tag.bird)`) which could be referenced if this enum was MATCHING.

I'm open to making this just MATCHING, but it's most likely we'd just add AND in the nearer future. @AaronCasas